staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types_discovery.go

```diff
@@ -0,0 +1,143 @@
+	// Versions are the versions supported in this group.
+	// +listType=map
+	// +listMapKey=version
+	Versions []APIVersionDiscovery `json:"versions,omitempty" protobuf:"bytes,2,rep,name=versions"`
+}
+
+// APIVersionDiscovery holds a list of APIResourceDiscovery types that are served for a particular version within an API Group.
+type APIVersionDiscovery struct {
+	// Version is the name of the version within a group version.
+	Version string `json:"version" protobuf:"bytes,1,opt,name=version"`
+	// +listType=map
+	// +listMapKey=resource
+	Resources []APIResourceDiscovery `json:"resources,omitempty" protobuf:"bytes,2,rep,name=resources"`
+	// Most recently observed status of the APIVersionDiscovery.
+	// This data may not be up to date.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Status APIVersionDiscoveryStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// APIResourceDiscovery provides information about an API resource for discovery.
+type APIResourceDiscovery struct {
+	// Resource is the plural name of the resource.  This is used in the URL path and is the unique identifier
+	// for this resource across all versions.
+	// resources with non-"" groups are located at /apis/<APIGroupDiscovery.objectMeta.name>/<APIVersionDiscovery.version>/<APIResourceDiscovery.Resource>
+	// resource with "" groups are located at /api/v1/<APIResourceDiscovery.Resource>
+	Resource string `json:"resource" protobuf:"bytes,1,opt,name=resource"`
+	// ReturnType describes the type of serialization that will be returned from this endpoint.
+	ReturnType APIDiscoveryKind `json:"returnType" protobuf:"bytes,2,opt,name=returnType"`
+	// Scope indicates the scope of a resource, either Cluster or Namespace
+	Scope ResourceScope `json:"scope" protobuf:"bytes,3,opt,name=scope"`
+	// SingularName is the singular name of the resource.  This allows clients to handle plural and singular opaquely.
+	// The singularName is more correct for reporting status on a single item and both singular and plural are allowed
+	// from the kubectl CLI interface.
+	SingularName string `json:"singularName" protobuf:"bytes,4,opt,name=singularName"`
+	// Verbs is a list of supported kube verbs (this includes get, list, watch, create,
+	// update, patch, delete, deletecollection, other)
+	Verbs Verbs `json:"verbs" protobuf:"bytes,5,opt,name=verbs"`
+	// ShortNames is a list of suggested short names of the resource.
+	// +listType=set
+	ShortNames []string `json:"shortNames,omitempty" protobuf:"bytes,6,rep,name=shortNames"`
+	// Categories is a list of the grouped resources this resource belongs to (e.g. 'all')
+	// +listType=set
+	Categories []string `json:"categories,omitempty" protobuf:"bytes,7,rep,name=categories"`
+	// Subresources is a list of subresources provided by this resource.  Subresources are located at
```

Comment:
located at /apis/<APIGroupDiscovery.objectMeta.name>/<APIVersionDiscovery.version>/<APIResourceDiscovery.Resource>/name-of-instance/<APIResourceDiscovery.subresources[i].subresource>